docs/developer/designs/min-runtime/README.md

```diff
@@ -0,0 +1,118 @@
+# [DRAFT] Minimum runtime before preemptions and reclaims
+
+## Overview
+
+This document proposes a new feature called "reclaim-min-runtime" and "preempt-min-runtime" that provides configurable guarantees for job runtime before preemption or reclaim can occur. This feature enables administrators to define minimum runtime guarantees at various levels of the resource hierarchy: node pool, nth level queue and leaf-queue.
+
+## Motivation
+
+Unrestrained preemption can lead to resource thrashing, where jobs are repeatedly preempted before making meaningful progress. The min-runtime feature addresses this issue by providing configurable minimum runtime guarantees that can be set by either cluster operators or sometimes within parts of the queue to provide guarantees about minimal useful work.
+
+## Detailed Design
+
+### Reclaim Min Runtime Configuration
+
+The reclaim-min-runtime parameter can be configured with the following values:
+
+- **0 (default)**: Jobs are always preemptible via reclaims
+- **-1**: Jobs are never preemptible via reclaims
```

Comment:
I am not sure about "-1" value, it sounds like a way to transform a preemptible job into a non-preemptible one. We already have this option - maybe it is not really needed?

Reply:
There were discussions that this could be used instead of priority to denote non-preemptible, but we can leave it out if thats not the case.